db/system_keyspace.hh

```diff
@@ -24,6 +24,7 @@
 #include "locator/host_id.hh"
 #include "virtual_tables.hh"
 #include "types/types.hh"
+#include <boost/icl/interval_map.hpp>
```

Comment:
Please employ type erasure to avoid bringing this into scope of all includers.

Reply:
Although I see one in locator/abstract_replication_strategy.hh, I thought I cleaned those up.

Reply:
> Please employ type erasure to avoid bringing this into scope of all includers.

Fixed by pimpling away the interval_map to `system_keyspace.cc`.